pytensor/tensor/rewriting/math.py

```diff
@@ -1914,6 +1915,33 @@ def local_pow_canonicalize(fgraph, node):
         return [alloc_like(node.inputs[0], node.outputs[0], fgraph)]
 
 
+@register_canonicalize
+@node_rewriter([pt_pow])
+def local_pow_canonicalize_base_1(
+    fgraph: FunctionGraph, node: Apply
+) -> list[TensorVariable] | None:
+    """
+    Replace `1 ** x` with 1, broadcast to the shape of the output.
+
+    Parameters
+    ----------
+    fgraph: FunctionGraph
+        Full function graph being rewritten
+    node: Apply
+        Specific node being rewritten
+
+    Returns
+    -------
+    rewritten_output: list[TensorVariable] | None
+        Rewritten output of node, or None if no rewrite is possible
+    """
+    cst = get_underlying_scalar_constant_value(
+        node.inputs[0], only_process_constants=True, raise_not_constant=False
+    )
+    if cst == 1:
+        return [alloc_like(node.inputs[0], node.outputs[0], fgraph)]
```

Comment:
This could make an infinite recursion if the `ShapeOpt` is not running, as it will default to `alloc(1, *pow(1, x).shape)`. 

You can do `pt.broadcast_arrays(*node.inputs)[0]` instead?

Reply:
Sure, but does that mean we should also change `local_canonicalize_pow` ? Because I copied the return from there

Reply:
Lines to that rewrite?

Reply:
found it yes. Why don't you combine your changes with that rewrite?

Reply:
check how it looks now? I had to set the dtype to the output as well, not sure if there's a better way